docs/Kube-API.md

```diff
@@ -24,6 +24,15 @@ When the ISO is ready, an URL will be available in the CR.
 ### [NMStateConfig](https://github.com/openshift/assisted-service/blob/master/internal/controller/api/v1beta1/nmstate_config_types.go)
 The NMStateConfig contains network configuration that will applied on the hosts. See NMState repository [here](https://github.com/nmstate/nmstate).
 
+To link between an InstallEnv to NMState (either one or more):
+
+- InstallEnv CR: add a label to nmStateConfigLabelSelector with a user defined name and value.
+- NMState CR: Specify the same label + value in Object metadata.
+
+Upon InstallEnv creation, the InstallEnv controller will search by label+value for matching NMState resources and construct a config to be sent as StaticNetworkConfig as a part of ImageCreateParams. The backend does all validations, and currently, there is no handling of configuration conflicts (e.g., two nmstate resources using the same MAC address).
+
+The InstallEnv controller will watch for NMState config creation/changes and search for corresponding InstallEnv resources to reconcile since we need to regenerate the image for those.
```

Comment:
```suggestion
The InfraEnv controller will watch for NMState config creation/changes and search for corresponding InstallEnv resources to reconcile since we need to regenerate the image for those.
```

Reply:
ditto